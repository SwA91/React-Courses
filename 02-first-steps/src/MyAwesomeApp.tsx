import type { CSSProperties } from "react";

const firstName = "Pepe";
const lastName = "Fernandez";
const array = [1, 2, 3, 4];
const isActive = true;
const myobj = {
  pepe: "juan",
};
// protip para tener las propiedades del CSS
const myStyles: CSSProperties = {
  backgroundColor: "red",
  borderRadius: 10,
  padding: isActive ? 10 : 5,
};
export function MyAwesomeApp() {
  // las constantes mejor dejarlas fuera
  // porque las funciones son llamadas constantementes
  // esto no es HTML sino JSX
  return (
    <>
      <h1>{firstName}</h1>
      <h3>{lastName}</h3>
      <p>{array.join(",")}</p>
      <p>{2 + 2}</p>
      <h1>{isActive ? "activo" : "desactivado"}</h1>
      <p style={myStyles}>{JSON.stringify(myobj)}</p>
    </>
  );
}
