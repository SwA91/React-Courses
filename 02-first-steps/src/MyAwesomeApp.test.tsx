import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import { render, screen } from "@testing-library/react";

describe("MyAwesomeApp", () => {
  test("should render firstname and lastname", () => {
    // ARRANGE
    const { container } = render(<MyAwesomeApp />);
    // screen.debug();
    // vemos todo el html renderizado
    // console.log(container.innerHTML);

    // ACT
    const h1 = container.querySelector("h1"); // primer h1

    // ASSERT
    expect(h1?.innerHTML).toContain("Pepe");
  });
  test("should render with screen", () => {
    render(<MyAwesomeApp />);
    screen.findByRole("heading");
    //   luego mejoramos los test con screen
    // con los eventos es mejor usar screen y no solo container
  });

  test("should match snapshot", () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
    // es rigido el snapshot para las pruebas
    // si el componente recibe muchas modificaciones
    // mejor no usar snapshot
  });
});
