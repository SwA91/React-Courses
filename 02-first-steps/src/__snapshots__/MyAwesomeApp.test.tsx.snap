// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <h1>
    Pepe
  </h1>
  <h3>
    Fernandez
  </h3>
  <p>
    1,2,3,4
  </p>
  <p>
    4
  </p>
  <h1>
    activo
  </h1>
  <p
    style="background-color: red; border-radius: 10px; padding: 10px;"
  >
    {"pepe":"juan"}
  </p>
</div>
`;
